### PR TITLE
1676 - Allow '+' in IBDO Form name

### PR DESCRIPTION
**Pull request by aL118**
Added + to the regex that checks if valid form name and updated unit test. Also updated some comments.

### CODE REVIEW

File: src/test/java/emissary/util/PayloadUtilTest.java [hunk 1]
```diff
@@ -243,14 +243,15 @@ void testXmlSerizliaztion() {
     void testIsValidForm() {
         // Check that all expected valid characters are valid
         String alphaLow = "abcdefghijklmnopqrstuvwxyz";
-        assertTrue(PayloadUtil.isValidForm(alphaLow), "Lower case alpha characters are not considered valid");
-        assertTrue(PayloadUtil.isValidForm(alphaLow.toUpperCase()), "Upper case alpha characters are not considered valid");
-        assertTrue(PayloadUtil.isValidForm("0123456789"), "Numeric characters are not considered valid");
-        assertTrue(PayloadUtil.isValidForm("-_"), "Dash and underscore aren't considered valid");
-        assertTrue(PayloadUtil.isValidForm("formName-(suffixInParens)"), "Parentheses aren't considered valid");
-        assertTrue(PayloadUtil.isValidForm("formName-(application/xml)"), "Slash aren't considered valid");
+        assertTrue(PayloadUtil.isValidForm(alphaLow), "Lower case alpha characters are valid");
```
Comment (drivenflywheel) [inline src/test/java/emissary/util/PayloadUtilTest.java:246]: I think we should alter each messages in this test to more clearly indicate what the expected result is. The original and proposed messages are ambiguous regarding whether they're stating desired vs observed behavior, but we can take away the ambiguity:
```suggestion
        assertTrue(PayloadUtil.isValidForm(alphaLow), "Lower case alpha characters are expected to be valid");
```
or 
```suggestion
        assertTrue(PayloadUtil.isValidForm(alphaLow), "Expected lower case alpha characters to be valid");
```

File: src/test/java/emissary/util/PayloadUtilTest.java [hunk 2]
```diff
@@ -243,14 +243,15 @@ void testXmlSerizliaztion() {
...
+        assertTrue(PayloadUtil.isValidForm(alphaLow.toUpperCase()), "Upper case alpha characters are valid");
+        assertTrue(PayloadUtil.isValidForm("0123456789"), "Numeric characters are valid");
+        assertTrue(PayloadUtil.isValidForm("-_"), "Dash and underscore are valid");
+        assertTrue(PayloadUtil.isValidForm("formName-(suffixInParens)"), "Parentheses are valid");
+        assertTrue(PayloadUtil.isValidForm("formName-(application/xml)"), "Slash are valid");
         assertFalse(PayloadUtil.isValidForm("."), "Dot isn't considered valid");
         assertFalse(PayloadUtil.isValidForm(" "), "Space isn't considered valid");
+        assertTrue(PayloadUtil.isValidForm("+"), "Plus are valid");
```
Comment (drivenflywheel) [inline src/test/java/emissary/util/PayloadUtilTest.java:254]:
```suggestion
        assertTrue(PayloadUtil.isValidForm("+"), "'+' is expected to be a valid form character");
```

**Review by drivenflywheel — CHANGES_REQUESTED** [review]
Looks good overall.  Would like to make some assertion messages less ambiguous.

**Comment by aL118** [comment]
Forgot to add closes [#502](https://github.com/NationalSecurityAgency/burrito-grande/issues/1676)
Also just realized that title had the wrong issue number so it's been fixed.

**Review by ldhardy — APPROVED** [review]
Those previous test assertion messages were completely opposite of the expected behavior, huh?  That's confusing, glad it's fixed now.

**Review by drivenflywheel — APPROVED** [review]
Looks great.  Ready to ship. 🚀

**Pull request merged**